app/controllers/api_controller/generic.rb

```diff
@@ -130,5 +142,18 @@ def delete_resource_action(klass, type, id)
       log_result(result)
       result
     end
+
+    def set_ownership_action(klass, type, id, ownership)
+      if ownership.blank?
+        action_result(false, "Must specify a valid owner or group for setting ownership")
+      else
+        errors = klass.set_ownership([id], ownership)
+        details = ownership.each.collect { |key, obj| "#{key}: #{obj.name}" }.join(", ")
+        desc = "setting ownership of #{type} id #{id} to #{details}"
+        errors == true ? action_result(true, desc) : action_result(false, errors.values.join(", "))
```

Comment:
I found this a little confusing - if `errors == true` there are no errors? Perhaps `errors` could be renamed, or this logic restructured a bit?